src/middleware/devtools.ts

```diff
@@ -147,13 +148,50 @@ type DevtoolsImpl = <T>(
 
 export type NamedSet<T> = WithDevtools<StoreApi<T>>['setState']
 
+type ConnectResponse = ReturnType<
+  NonNullable<Window['__REDUX_DEVTOOLS_EXTENSION__']>['connect']
+>
+const connections: Map<string | undefined, ConnectResponse> = new Map()
+
+type ConnectionStoreApis = (StoreApi<any> & { store: string })[]
+const connectionStoreApis: Map<string | undefined, ConnectionStoreApis> =
+  new Map()
+type ConnectionInitialStoreStates = (any & { store: string })[]
+const connectionInitialStoreStates: Map<
+  string | undefined,
+  ConnectionInitialStoreStates
+> = new Map()
```

Comment:
Can this be just a single map?
```ts
type Name = string | undefined // `name` option
type Store = string | undefined // `store` option
type Connections = Map<Name, [ConnectResponse, Record<Store, StoreApi<unknown>>]>
```
Only having one store api where `store=undefined` is a special case.

---
Oh, we need to allow having multiple `name=undefined` instances? That seems tricky.
Then this? 🤔 
```ts
type Connections = Set<[Name, ConnectResponse, Record<Store, StoreApi<unknown>>]>
```

Reply:
Yeah, multiple name undefined instances is a default behavior which is not introduced by this pr. And it is not fully covered by tests